src/main/java/dev/thedocruby/resounding/Cache.java

```diff
@@ -282,17 +285,17 @@ public static Branch growOctree(WorldChunk chunk, Branch root) {
     // determined by temperature & humidity (global transmission coefficient -> alters permeability)
     public static double transmission = 1;
 
-    public static boolean generate(Consumer<String> logger) {
+    public static boolean generate(LogBuilder logger) {
         // FabricTagProvider.BlockTagProvider x = null;
-        logger.accept(Registry.BLOCK.getKey(Blocks.AIR).toString());
+        logger.log(Registry.BLOCK.getKey(Blocks.AIR));
         // Registry.BLOCK.forEach(
-        Registry.REGISTRIES.streamTags().forEach(
-                (tagKey) -> {
-                    logger.accept(
-                        tagKey.registry().getValue().getPath()
-                    );
-                }
-        );
+        logger.log("{}", () -> {
+            return Registry.REGISTRIES.streamTags()
+                    .map(TagKey::registry)
+                    .map(RegistryKey::getValue)
+                    .map(Identifier::getPath)
+                    .collect(Collectors.joining("\n"));
+        });
```

Comment:
While this code isn't of much consequence, given it's being refactored... Multiple maps have worse performance than a single iteration, and this degrades readability in this circumstance.

Reply:
I see. Still, I'll check if I could replace this (and some others) with a non-stream; or, a single map.